Adding support for ABI action_results

aaroncox opened the pull request:
Beyond the unit tests, I also set an ABI (the one from the unit tests) using this code on the `corecorecore` account.

Seems to work: https://jungle4.eosq.eosnation.io/account/corecorecore/abi

jnordberg reviewed (approved):
LGTM assuming the action_results thingy is a binary extension on the ABI type.

A test decoding from a known nodeos binary blob would be a good addition.

aaroncox commented:
Removed that comment, and also added a test where it's pushing an ABI and retrieving it (both JSON and raw) to decode. 

As for actual action_return values in push/send transaction calls, we'll have to handle that soon.
